lib/src/migrators/module.dart

```diff
@@ -174,27 +191,30 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
     // they should be forwarded or not.
     var shown = <Uri, Set<String>>{};
     var hidden = <Uri, Set<String>>{};
-    categorizeMember(Uri url, String originalName, String newName) {
+    categorizeMember(
+        SassNode declaration, String originalName, String newName) {
+      var url = declaration.span.sourceUrl;
       if (url == _currentUrl) return;
-      if (_shouldForward(originalName)) {
+      if (_shouldForward(originalName) && !originalName.startsWith('-')) {
         shown[url] ??= {};
         shown[url].add(newName);
-      } else {
+      } else if (!newName.startsWith('-') && !newName.startsWith(r'$-')) {
         hidden[url] ??= {};
         hidden[url].add(newName);
       }
     }
 
-    for (var entry in _scope.variables.entries) {
-      if (entry.value is! VariableDeclaration) continue;
-      categorizeMember(entry.value.span.sourceUrl,
-          (entry.value as VariableDeclaration).name, '\$${entry.key}');
+    for (var node in _scope.variables.values) {
```

Comment:
I'm surprised that you still need to track the scope here. Does `References` not provide enough context?

Reply:
Okay, I think I found a way to eliminate the need for scope tracking in the main migration pass, with the exception of tracking unreferencable members from nested imports. Would it be better to move this detection to `_ReferenceVisitor` or keep tracking just these members in the main visitor? Or can you think of a way to detect these references without tracking the scope?

Reply:
I think I would just track this in `References`—exposed through something like `Set<SassNode> get fromNestedImports`.

Reply:
I ended up refactoring the unreferencable-members parts of `Scope` into a separate `UnreferencableMembers` class, using the regular `Scope` in `_ReferenceVisitor` and `UnreferencableMembers` in the main visitor.

The issue with tracking it in `References` was that we need ensure both that we don't reference members from nested imports in the importing stylesheet and that we don't reference local members from the importing stylesheet within the nested import (or its dependencies). I couldn't find a way to get both without duplicating a lot of logic between the two visitors.

I also prefer that `References` represent only the current state of the stylesheets being migrated to make it potentially reusable for future migrators/static analysis, so it felt weird having it throw exceptions for unreferencable members when they don't actually become unreferencable until the nested import is migrated to `load-css`.